devicetypes/smartthings/zigbee-motion-temp-humidity-sensor.src/zigbee-motion-temp-humidity-sensor.groovy

```diff
@@ -0,0 +1,280 @@
+		// OR we have received the same reading twice in a row
+		// OR we don't currently have a battery reading
+		// OR the value we just received is at least 2 steps off from the last reported value
+		if (state?.lastVolts == null || state?.lastVolts == volts || device.currentState("battery")?.value == null || Math.abs(curValVolts - volts) > 0.1) {
+			def pct = (volts - minVolts) / (maxVolts - minVolts)
+			def roundedPct = Math.round(pct * 100)
+			if (roundedPct <= 0)
+				roundedPct = 1
+			result.value = Math.min(100, roundedPct)
+		} else {
+			// Don't update as we want to smooth the battery values, but do report the last battery state for record keeping purposes
+			result.value = device.currentState("battery").value
+		}
+		result.descriptionText = "${device.displayName} battery was ${result.value}%"
+		state.lastVolts = volts
+	}
+
+	return result
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+	log.debug "Battery Percentage rawValue = ${rawValue} -> ${rawValue / 2}%"
+	def result = [:]
+
+	if (0 <= rawValue && rawValue <= 200) {
+		result.name = 'battery'
+		result.translatable = true
+		result.value = Math.round(rawValue / 2)
+		result.descriptionText = "${device.displayName} battery was ${result.value}%"
+	}
+
+	return result
+}
+
+private Map getMotionResult(value) {
+	log.debug 'motion'
+	String descriptionText = value == 'active' ? "${device.displayName} detected motion" : "${device.displayName} motion has stopped"
+	return [
+		name           : 'motion',
+		value          : value,
+		descriptionText: descriptionText,
+		translatable   : true
+	]
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+		zigbee.readAttribute(0x0405, 0x0000) +
+		zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000) +
+		zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+		zigbee.enrollResponse()
+
+	return refreshCmds
+}
+
+def configure() {
+	// Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+	// enrolls with default periodic reporting until newer 5 min interval is confirmed
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "Configuring Reporting"
+	
+	// temperature minReportTime 30 seconds, maxReportTime 5 min. Reporting interval if no activity
+	// battery minReport 30 seconds, maxReportTime 6 hrs by default
+	// humidity minReportTime 30 seconds, maxReportTime 60 min
+	def configCmds = []
+
+	configCmds += zigbee.batteryConfig() +
+		zigbee.temperatureConfig(30, 300) +
+		zigbee.configureReporting(0x0405, 0x0000, DataType.UINT16, 30, 3600,100)
```

Comment:
You can use `zigbee.RELATIVE_HUMIDITY_CLUSTER` instead of 0x0405 here and the other places it's referenced

Reply:
nit: space before `100`